shell/browser/native_window_mac.mm

```diff
@@ -469,7 +473,25 @@ void ReorderChildWindowAbove(NSWindow* child_window, NSWindow* other_window) {
     return;
 
   if (focus) {
-    [[NSApplication sharedApplication] activateIgnoringOtherApps:NO];
+    // If we're a panel window, we do not want to activate the app,
+    // which enables Electron-apps to build Spotlight-like experiences.
+    //
+    // On macOS < Sonoma, "activateIgnoringOtherApps:NO" would not
+    // activate apps if focusing a window that is inActive. That
+    // changed with macOS Sonoma.
+    //
+    // There's a slim chance we should have never called
+    // activateIgnoringOtherApps, but we tried that many years ago
+    // and saw weird focus bugs on other macOS versions. So, to make
+    // this safe, we're gating by versions.
+    if (@available(macOS 14.0, *)) {
+      if (!IsPanel(window_)) {
```

Comment:
I checked with a minimal Cocoa sample app, and it seems to me that simply switching the deprecated `activateIgnoringOtherApps:` API with the new `activate` API solves the issue. No need to gate it based on `IsPanel`.

So, can you try it without this check?